llvm/lib/SYCL/VXXIRDowngrader.cpp

```diff
@@ -266,10 +229,15 @@ struct VXXIRDowngrader : public ModulePass {
       I->eraseFromParent();
   }
 
-  void convertPoinsonToZero(Module &M) {
-    for (auto &PV : M.getContext().pImpl->PVConstants)
+  void convertPoisonToZero(Module &M) {
+    for (auto &PV : M.getContext().pImpl->PVConstants) {
+      Type* Ty = PV.second->getType();
+      if (PV.second->use_empty() || Ty->isVoidTy() || Ty->isFunctionTy() ||
+          Ty->isLabelTy() || Ty->isMetadataTy())
+        continue;
       PV.second.get()->replaceAllUsesWith(
           Constant::getNullValue(PV.second.get()->getType()));
+    }
   }
 
   void removeMetaDataValues(Module &M) {
```

Comment:
You cannot use somehow your new `removeMetadata()`?

Reply:
This function removes values of type metadata because they are not supported by the HLS backend.

removeMetadata in SYCLUtils remove a specific metadata from the global scope.

Reply:
I guess the problem is about naming so I do not understand.
Are you talking about https://llvm.org/docs/LangRef.html#metadata-type ?
In the comment it could be clearer with
```
/// This function removes metadata type values because they are not
```

Reply:
Yes this is what this function removes. to be more precise this function removes the llvm intrinsics that use metadata typed arguments. such that there is not metadata typed value in the IR.

Reply:
Now that you have written explanation you win the right to put it in a comment ;-)